pkg/controller/util/endpoint/controller_utils_test.go

```diff
@@ -224,3 +232,117 @@ func TestShouldPodBeInEndpoints(t *testing.T) {
 		}
 	}
 }
+
+func TestGetPodServiceMemberships(t *testing.T) {
```

Comment:
This may provide more flexibility with test cases if you had a set of test cases containing the following:
- pod
- services
- expectedServiceMemberships

A simple pod and/or service generator might also simplify this workflow. This would allow a variety of more complex test cases to be added relatively easily.